pulser-simulation/pulser_simulation/simulation.py

```diff
@@ -157,6 +157,13 @@ def __init__(
             sampling_rate,
             noise_model,
         )
+        self._noiseless_hamiltonian = Hamiltonian(
+            self.samples_obj,
+            self._register.qubits,
+            device,
+            sampling_rate,
+            NoiseModel(),
+        )
```

Comment:
This can take a bit of time, might be better to calculate it and cache it only when it is first requested in `get_hamiltonian()`. A convenient way to do so is to define  `_noiseless_hamiltonian` as a `cached_property`